### PR TITLE
[refactor] #4167: asset subcommand in client CLI to be consistent

### LINKED ISSUE #4167: [suggestion] Make Assets API of Iroha Client CLI consistent
I find args naming used by different `asset` subcommands confusing:

- `mint`, `burn`, `get`:
    - `--account`: `AccountId` like `name@domain_name`
    - `--asset`: `AssetDefinitionId` like `name#domain_name`
- `register`:
    - `--id`: `AssetDefinitionId` like `name#domain_name`
- `transfer`:
    - `--from`, `--to`: `AccountId` like `name@domain_name`
    - `--asset-id`: `AssetDefinitionId` like `name#domain_name`

Suggestions:

- For `mint`/`burn`/`get`, use composite `--asset-id`: `AssetId` (like `asset##account@domain`)
- For `register`, use `--definition-id`, not just `--id`
- For `transfer`, use `--asset-id`: `AssetId` (like `asset##account@domain`), and `--to`. `--from` is not needed, since the original account is specified as part of the `--asset-id`

<details>

<summary>Help messages of each command</summary>

**`mint`**
```
Command for minting asset in existing Iroha account

Usage: iroha_client_cli asset mint [OPTIONS] --account <ACCOUNT> --asset <ASSET> --quantity <QUANTITY>

Options:
      --account <ACCOUNT>    Account id where asset is stored (in form of `name@domain_name')
      --asset <ASSET>        Asset id from which to mint (in form of `name#domain_name')
  -q, --quantity <QUANTITY>  Quantity to mint
  -m, --metadata <PATH>      The JSON/JSON5 file with key-value metadata pairs
  -h, --help                 Print help
```
**`burn`**
```
Command for burning asset in existing Iroha account

Usage: iroha_client_cli asset burn [OPTIONS] --account <ACCOUNT> --asset <ASSET> --quantity <QUANTITY>

Options:
      --account <ACCOUNT>    Account id where asset is stored (in form of `name@domain_name')
      --asset <ASSET>        Asset id from which to mint (in form of `name#domain_name')
  -q, --quantity <QUANTITY>  Quantity to mint
  -m, --metadata <PATH>      The JSON/JSON5 file with key-value metadata pairs
  -h, --help                 Print help
```

**`get`**

```
Get info of asset

Usage: iroha_client_cli asset get --account <ACCOUNT> --asset <ASSET>

Options:
      --account <ACCOUNT>  Account where asset is stored (in form of `name@domain_name')
      --asset <ASSET>      Asset name to lookup (in form of `name#domain_name')
  -h, --help               Print help
```
**`register`**
```
Register subcommand of asset

Usage: iroha_client_cli asset register [OPTIONS] --id <ID> --value-type <VALUE_TYPE>

Options:
  -i, --id <ID>                  Asset id for registering (in form of `name#domain_name')
  -u, --unmintable               Mintability of asset
  -v, --value-type <VALUE_TYPE>  Value type stored in asset
  -m, --metadata <PATH>          The JSON/JSON5 file with key-value metadata pairs
  -h, --help                     Print help

```
**`transfer`**
```
Transfer asset between accounts

Usage: iroha_client_cli asset transfer [OPTIONS] --from <FROM> --to <TO> --asset-id <ASSET_ID> --quantity <QUANTITY>

Options:
  -f, --from <FROM>          Account from which to transfer (in form `name@domain_name')
  -t, --to <TO>              Account to which to transfer (in form `name@domain_name')
  -a, --asset-id <ASSET_ID>  Asset id to transfer (in form like `name#domain_name')
  -q, --quantity <QUANTITY>  Quantity of asset as number
  -m, --metadata <PATH>      The JSON/JSON5 file with key-value metadata pairs
  -h, --help                 Print help

```

</details>

### PR DESCRIPTION
**Pull request by Asem-Abdelhady**
## Description

In asset subcommand in the CLI operations like `mint`, `burn`, `get`, `transfer` had `register`. I refactored the subcommand to follow the suggestions in #4167.

### Linked issue

Closes #4167

#### Register
```bash
./iroha_client_cli asset  --definition-id="tea#looking_glass" --value-type=Quantity 
```
#### Mint
```bash
./iroha_client_cli asset mint --asset-id="tea##mad_hatter@looking_glass" --quantity="100"
```
#### Burn
```bash
./iroha_client_cli asset burn --asset-id="tea##mad_hatter@looking_glass" --quantity="10"
```

#### Get
```bash
./iroha_client_cli asset get --asset-id="tea##mad_hatter@looking_glass"  
```
#### Transfer
```bash
./iroha_client_cli asset transfer --to="white_rabbit@looking_glass" --asset-id="tea##mad_hatter@looking_glass" --quantity="5"  
```
### Checklist

- [X] I've read `CONTRIBUTING.md`
- [X] I've used the standard signed-off commit format (or will squash just before merging)
- [X] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [X] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: client_cli/src/main.rs [hunk 1]
```diff
@@ -830,15 +819,12 @@ mod asset {
     /// Transfer asset between accounts
     #[derive(clap::Args, Debug)]
     pub struct Transfer {
-        /// Account from which to transfer (in form `name@domain_name')
-        #[arg(short, long)]
-        pub from: AccountId,
         /// Account to which to transfer (in form `name@domain_name')
         #[arg(short, long)]
         pub to: AccountId,
```
Comment (Arjentix) [inline client_cli/src/main.rs:824]: I think `short` can be removed here as well

**Review by Arjentix — APPROVED** [review]

**Comment by Arjentix** [comment]
Need to make CI pass fine

**Comment by Asem-Abdelhady** [comment]
> Need to make CI pass fine

Yes, sure for that I would contact Alex when gets better.

File: client_cli/src/main.rs (hunk 1 above)
Comment (Asem-Abdelhady) [reply client_cli/src/main.rs:824]: I think it should also be removed for `AssetDefinitionId` in `Register`

**Review by VAmuzing — APPROVED** [review]
Seems like the help section for the commands in #4167 should also be updated to reflect removal of 'from' parameters

**Comment by Asem-Abdelhady** [comment]
> Seems like the help section for the commands in #4167 should also be updated to reflect removal of 'from' parameters

yes, i will do that

File: client_cli/src/main.rs [hunk 2]
```diff
@@ -797,12 +792,9 @@ mod asset {
     /// Command for minting asset in existing Iroha account
     #[derive(clap::Args, Debug)]
     pub struct Burn {
-        /// Account id where asset is stored (in form of `name@domain_name')
+        /// Asset id for the asset (in form of `asset##account@domain_name')
```
Comment (0x009922) [inline client_cli/src/main.rs:795]:
```suggestion
        /// Asset id for the asset (in form of `asset##account@domain_name`)
```

File: client_cli/src/main.rs [hunk 3]
```diff
@@ -830,15 +819,12 @@ mod asset {
...
-        /// Account from which to transfer (in form `name@domain_name')
-        #[arg(short, long)]
-        pub from: AccountId,
         /// Account to which to transfer (in form `name@domain_name')
         #[arg(short, long)]
         pub to: AccountId,
-        /// Asset id to transfer (in form like `name#domain_name')
-        #[arg(short, long)]
-        pub asset_id: AssetDefinitionId,
+        /// Asset id to transfer (in form like `asset##account@domain_name')
```
Comment (0x009922) [inline client_cli/src/main.rs:825]:
```suggestion
        /// Asset id to transfer (in form like `asset##account@domain_name`)
```

**Review by Arjentix — APPROVED** [review]

**Review by VAmuzing — APPROVED** [review]

**Pull request merged**